src/test/ui/associated-types/defaults-specialization.stderr

```diff
@@ -0,0 +1,50 @@
+error[E0053]: method `make` has an incompatible type for trait
+  --> $DIR/defaults-specialization.rs:17:18
+   |
+LL |     fn make() -> Self::Ty;
+   |                  -------- type in trait
+...
+LL |     fn make() -> u8 { 0 }
+   |                  ^^ expected associated type, found u8
+   |
+   = note: expected type `fn() -> <A<T> as Tr>::Ty`
+              found type `fn() -> u8`
+
+error[E0053]: method `make` has an incompatible type for trait
+  --> $DIR/defaults-specialization.rs:33:18
+   |
+LL |     fn make() -> Self::Ty;
+   |                  -------- type in trait
+...
+LL |     fn make() -> bool { true }
+   |                  ^^^^ expected associated type, found bool
+   |
+   = note: expected type `fn() -> <B<T> as Tr>::Ty`
+              found type `fn() -> bool`
+
+error[E0308]: mismatched types
+  --> $DIR/defaults-specialization.rs:24:29
+   |
+LL |     fn make() -> Self::Ty { 0u8 }
+   |                  --------   ^^^ expected associated type, found u8
+   |                  |
+   |                  expected `<A2<T> as Tr>::Ty` because of return type
+   |
+   = note: expected type `<A2<T> as Tr>::Ty`
+              found type `u8`
```

Comment:
I would love to see a more general way to expand the wording for these to help people understand how to actually _return_ a value that conforms to the associated type. This is super confusing even for people with intermediate knowledge of the language.